### PR TITLE
ContentMigrationBatch#S32

### PR DESCRIPTION
**Pull request by simgobb**
https://developers.google.com/web/updates/2012/06/How-to-convert-ArrayBuffer-to-and-from-String
https://developers.google.com/web/updates/2012/07/Arrived-xhr-send-ArrayBufferView
https://developers.google.com/web/updates/2012/07/Drag-and-drop-a-folder-onto-Chrome-now-available
https://developers.google.com/web/updates/2012/07/How-to-measure-browser-graphics-performance
https://developers.google.com/web/updates/2012/07/Taking-advantage-of-GPU-acceleration-in-the-2D-canvas

@rachelandrew

### CODE REVIEW

File: site/en/blog/how-to-measure-browser-graphics-performance/index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+description: >
+  How to measure browser graphics performance.
+authors:
+  - ilmariheikkinen
+date: 2012-07-22
+updated: 2019-02-21
+
+---
+
+Benchmarking browser graphics in a nutshell: draw as much as you can while maintaining a smooth frame rate. Once your framerate drops, you know how much you can draw per frame. End of post. ...No? Ok, I'll explain some more.
```
Comment (rachelandrew) [inline site/en/blog/how-to-measure-browser-graphics-performance/index.md:13]:
```suggestion
Benchmarking browser graphics in a nutshell: draw as much as you can while maintaining a smooth frame rate. Once your framerate drops, you know how much you can draw per frame. End of post. No? Ok, I'll explain some more.
```

File: site/en/blog/how-to-measure-browser-graphics-performance/index.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## How to do it then
+
+A better way to benchmark is to use a realistic drawing load and multiply it until the frame rate starts to chug. For example, if you’re writing a top-down game with a tilemap terrain, try drawing the tilemap every frame and seeing if it runs at 60 FPS. If yes, increase load (draw tilemap twice every frame, with a clear in between). Continue increasing until the FPS drops to a new stable level. Now you know how many layers of tilemap you can draw per frame.
+
+{% YouTube id="hAzhayTnhEI" %}
+
+
+Different graphics applications have different needs, so you should write your benchmarks with that in mind. Measure the graphics features that you are using in your app. When you find a slow scenario, try to reduce it down to the smallest piece of code that reproduces it (and file a bug report at "new.crbug.com":http://new.crbug.com if it should be faster.)
```
Comment (rachelandrew) [inline site/en/blog/how-to-measure-browser-graphics-performance/index.md:66]:
```suggestion
Different graphics applications have different needs, so you should write your benchmarks with that in mind. Measure the graphics features that you are using in your app. When you find a slow scenario, try to reduce it down to the smallest piece of code that reproduces it (and file a bug report at [new.crbug.com](https://new.crbug.com] if it should be faster.)
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]

**Comment by simgobb** [comment]
@rachelandrew I have noticed that the PR linting is failing on the URL that got changed on the last review. Not sure what's wrong though. Looks formatted correctly?

File: site/en/blog/how-to-measure-browser-graphics-performance/index.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+## How to do it then
+
+A better way to benchmark is to use a realistic drawing load and multiply it until the frame rate starts to chug. For example, if you’re writing a top-down game with a tilemap terrain, try drawing the tilemap every frame and seeing if it runs at 60 FPS. If yes, increase load (draw tilemap twice every frame, with a clear in between). Continue increasing until the FPS drops to a new stable level. Now you know how many layers of tilemap you can draw per frame.
+
+{% YouTube id="hAzhayTnhEI" %}
+
+
+Different graphics applications have different needs, so you should write your benchmarks with that in mind. Measure the graphics features that you are using in your app. When you find a slow scenario, try to reduce it down to the smallest piece of code that reproduces it (and file a bug report at [new.crbug.com](https://new.crbug.com] if it should be faster.)
```
Comment (rachelandrew) [inline site/en/blog/how-to-measure-browser-graphics-performance/index.md:66]:
```suggestion
Different graphics applications have different needs, so you should write your benchmarks with that in mind. Measure the graphics features that you are using in your app. When you find a slow scenario, try to reduce it down to the smallest piece of code that reproduces it (and file a bug report at [new.crbug.com](https://new.crbug.com) if it should be faster.)
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
the problem was an incorrect bracket, I've changed it and I think that should now pass the linter if you commit that.

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**